### PR TITLE
Added Dark

### PR DESCRIPTION
**Pull request by PonchoBT**

### CODE REVIEW

**Comment by miladvafaeifard** [comment]
DarkMode should be in components since this is a component not how to layout

File: src/layouts/DarkMode.astro [hunk 1]
```diff
@@ -0,0 +1,93 @@
+---
+const isDarkMode = new Date().getHours() > 18 || new Date().getHours() < 6;
+---
+
+
+  <head>
```
Comment (miladvafaeifard) [inline src/layouts/DarkMode.astro:6]: head tag is not needed here

File: src/layouts/DarkMode.astro [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    .icon.sun {
+      left: 12px;
+    }
+
+    .icon.moon {
+      right: 12px;
+    }
+      </style>
+  </head>
+  <body class={isDarkMode ? "dark" : ""}>
```
Comment (miladvafaeifard) [inline src/layouts/DarkMode.astro:66]: no body should be here since it is a component

File: src/layouts/DarkMode.astro [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    <div class="toggle-container">
+      <div class="toggle-button {isDarkMode ? 'dark' : ''}" id="toggle-button">
+        <div class="toggle-circle"></div>
+        <svg class="icon sun" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
+          <circle cx="12" cy="12" r="5"/>
+          <path d="M12 1v2M12 21v2M4.22 4.22l1.42 1.42M18.36 18.36l1.42 1.42M1 12h2M21 12h2M4.22 19.78l1.42-1.42M18.36 5.64l1.42-1.42"/>
+        </svg>
+        <svg class="icon moon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
+          <path d="M21 12.79A9 9 0 0111.21 3a7 7 0 000 14 9 9 0 009.79-4.21z"/>
+        </svg>
```
Comment (miladvafaeifard) [inline src/layouts/DarkMode.astro:76]: should create components for svg, see how svg components are organized

File: src/layouts/DarkMode.astro [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      </div>
+    </div>
+  
+    <script>
+      document.addEventListener("DOMContentLoaded", function () {
```
Comment (miladvafaeifard) [inline src/layouts/DarkMode.astro:81]: we have a library Alpinejs, we could use this

File: src/styles/global.css [hunk 5]
```diff
@@ -229,3 +229,28 @@ hr {
...
+body.dark span,
+body.dark li,
+body.dark ul,
+body.dark ol,
+body.dark footer,
+body.dark header,
+body.dark nav,
+body.dark .toggle-button,
+body.dark .DocSearch-Button {
+  background: rgb(58, 58, 58);
```
Comment (miladvafaeifard) [inline src/styles/global.css:253]: create a color var for 58 58 58

File: src/layouts/DarkMode.astro (hunk 2 above)
Comment (PonchoBT) [reply src/layouts/DarkMode.astro:66]: Added in component?

Comment (PonchoBT) [reply src/layouts/DarkMode.astro:66]: Oh, yes correct in component

File: src/styles/global.css (hunk 5 above)
Comment (PonchoBT) [reply src/styles/global.css:253]: ok

File: src/layouts/DarkMode.astro (hunk 4 above)
Comment (PonchoBT) [reply src/layouts/DarkMode.astro:81]: I don't understand, can you explain it to me please?

**Comment by miladvafaeifard** [comment]
no longer needed, there is another new PR: https://github.com/signlanguagetech/signlanguagetech.github.io/pull/21

**Pull request closed without merging**